common/djangoapps/third_party_auth/models.py

```diff
@@ -140,6 +140,13 @@ class ProviderConfig(ConfigurationModel):
             "authentication using the correct link is still possible."
         ),
     )
+    drop_existing_session = models.BooleanField(
+        default=False,
+        help_text=_(
+            "Whether to drop an existing session when accessing certain views and including "
```

Comment:
Could do. I haven't had time to do research on exactly which views might expect a tpa-hint parameter and how common the use is. Since this decorator logs the user out regardless as long as that parameter is present, I'm not sure what the overall implication would be if we applied it to the entire platform.